rendermodules/mesh_lens_correction/MeshAndSolveTransform.py

```diff
@@ -11,616 +11,569 @@
+            problem creating a mesh lens correction"""
+    pass
+
+
+def get_tspecs_and_matches(render, collection, stack, z, sectionId):
+    # load the specs and matches
+    tspecs = np.array(
+            renderapi.tilespec.get_tile_specs_from_z(
+                stack,
+                z,
+                render=render))
+    stack_ids = np.array([t.tileId for t in tspecs])
+
+    matches = renderapi.pointmatch.get_matches_within_group(
+            collection,
+            sectionId,
+            render=render)
+    match_ids = np.array(
+                [[m['pId'], m['qId']] for m in matches]).flatten()
+
+    unique_ids = np.intersect1d(stack_ids, match_ids)
+    if unique_ids.size == 0:
+        raise MeshLensCorrectionException(
+                "no tileId overlap between stack %s and collection %s" %
+                (stack, collection))
+
+    utspecs = []
+    for uid in unique_ids:
+        ind = np.argwhere(stack_ids == uid).flatten()[0]
+        utspecs.append(tspecs[ind])
+
+    umatches = []
+    for m in matches:
+        if (m['pId'] in unique_ids) & (m['qId'] in unique_ids):
+            umatches.append(m)
+
+    return utspecs, umatches
+
+
+def condense_coords(matches):
+    # condense point match structure into Nx2
+    x = []
+    y = []
+    for m in matches:
+        x += m['matches']['p'][0]
+        x += m['matches']['q'][0]
+        y += m['matches']['p'][1]
+        y += m['matches']['q'][1]
+    coords = np.transpose(np.vstack((np.array(x), np.array(y))))
+    return coords
+
+
+def even_distribution(coords, tile_width, tile_height, n):
+    # n: area divided into nxn
+    min_count = np.Inf
+    for i in range(n):
+        r = np.arange(
+                (i*tile_width/n),
+                ((i+1)*tile_width/n))
+        for j in range(n):
+            c = np.arange(
+                    (j*tile_height/n),
+                    ((j+1)*tile_height/n))
+            ind = np.argwhere(
+                    (coords[:, 0] >= r.min()) &
+                    (coords[:, 0] <= r.max()) &
+                    (coords[:, 1] >= c.min()) &
+                    (coords[:, 1] <= c.max())).flatten()
+            if ind.size < min_count:
+                min_count = ind.size
+    new_coords = []
+    for i in range(n):
+        r = np.arange(
+                (i*tile_width/n),
+                ((i+1)*tile_width/n))
+        for j in range(n):
+            c = np.arange(
+                    (j*tile_height/n),
+                    ((j+1)*tile_height/n))
+            ind = np.argwhere(
+                    (coords[:, 0] >= r.min()) &
+                    (coords[:, 0] <= r.max()) &
+                    (coords[:, 1] >= c.min()) &
+                    (coords[:, 1] <= c.max())).flatten()
+            a = np.arange(ind.size)
+            np.random.shuffle(a)
+            ind = ind[a[0:min_count]]
+            new_coords.append(coords[ind])
+    return np.concatenate(new_coords)
+
+
+def write_montage_qc_json(args, tilespecs):
```

Comment:
This function seems unnecessary -- could this just be written as a dictionary and output after validating an output schema?